### PR TITLE
Rename angularX configuration option to angular

### LINKED ISSUE #19354: Rename angularX configuration option to angular.
##### **Overview of the feature request**

Cleanup.

##### **Related issues or PR**

- [ ] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by qmonmert**
Fix #19354 

---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

File: jdl/jhipster/client-framework-types.js [hunk 1]
```diff
@@ -18,7 +18,7 @@
  */
 
 const Types = {
-  ANGULAR_X: 'angularX',
+  ANGULAR_X: 'angular',
```
Comment (DanielFran) [inline jdl/jhipster/client-framework-types.js:21]: Constant should be removed and keep only ANGULAR

Comment (qmonmert) [reply jdl/jhipster/client-framework-types.js]: yes

File: jdl/jhipster/application-options.js [hunk 2]
```diff
@@ -156,7 +156,7 @@ const optionValues = {
     [NO_CACHE_PROVIDER]: NO_CACHE_PROVIDER,
   },
   [optionNames.CLIENT_FRAMEWORK]: {
-    [ANGULAR_X]: ANGULAR_X,
+    [ANGULAR]: ANGULAR,
```
Comment (mshima) [inline jdl/jhipster/application-options.js:159]:
```suggestion
```

**Comment by mshima** [comment]
Tagging as major since jdl will break with angularX value.

**Comment by JusticeN** [comment]
For backward compatibility it will be better to allow both angular and angularx for now.
Update the documentation and and show warning when used. Then completely remove angularx in a later version

**Comment by DanielFran** [comment]
@JusticeN The target is for a the v8

File: jdl/jhipster/application-options.js (hunk 2 above)
Comment (DanielFran) [inline jdl/jhipster/application-options.js:159]:
```suggestion
```

**Review by DanielFran — CHANGES_REQUESTED** [review]

**Comment by Tcharl** [comment]
Thus is it possible to create the V8 branch?

File: jdl/jhipster/application-options.js (hunk 2 above)
Comment (qmonmert) [reply jdl/jhipster/application-options.js:159]: done

Comment (qmonmert) [reply jdl/jhipster/application-options.js:159]: done

**Comment by mraible** [comment]
This PR needs a rebase.

**Review by mshima — APPROVED** [review]
@qmonmert can you fix the conflicts and merge?

**Pull request merged**